dgbpy/hdf5.py

```diff
@@ -249,33 +249,36 @@ def getLogDir( withtensorboard, examplenm, platform, basedir, clearlogs, args ):
   logdir = logdir / Path(jobnm+str(nrsavedruns+1)+'_'+'m'.join( datetime.now().isoformat().split(':')[:-1] ))
   return logdir
 
-def getOutdType( classinfo ):
-  max = classinfo.max()
-  min = classinfo.min()
-  min_abs = np.abs(min)
-  max_abs = np.abs(max)
-  if max_abs>min_abs:
-    pass
+def getOutdType( classinfo, withunlabaled='No' ):
+  if withunlabaled == 'Yes':
+    return np.int16
```

Comment:
This is too much of a shortcut. If the labels contain [-1, 0, 1, 100000], np.int16 is not a good data type.
You should be able to keep most of logic but ajusting min at the very start:
if hasunlabels:
  min = np.min( [min, -1] )

Note that this assumes that unlabeled values are always mapped to -1, which may or may not be true.